File: _wpf/wpf-business-programmer/2-wpf-business-programmer-m2-exercise-files/Samples/VB/WPFListBox-VB/WPFComponents/MyCustomers.vb
Imports System
Imports System.Collections.Generic
Imports System.Data
Imports System.Data.SqlClient
Imports System.Linq

Public Class MyCustomers
  Inherits List(Of MyCustomer)

  Public Function GetCustomerList() As IEnumerable(Of MyCustomer)
    Dim dc As New AdvWorksDataContext()

    Dim items = From cust In dc.Customers _
                       Order By cust.LastName _
                       Select New MyCustomer()

    Return items
  End Function
End Class

File: _wpf/wpf-business-programmer/2-wpf-business-programmer-m2-exercise-files/Samples-2012/CSharp/WPFListBox/WPFComponents/MyCustomers.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.SqlClient;
using System.Linq;

namespace WPFComponents
{
  public class MyCustomers : List<MyCustomer>
  {
    public IEnumerable<MyCustomer> GetCustomerList()
    {
      AdvWorksDataContext dc = new AdvWorksDataContext();

      var items = from cust in dc.Customers
                  orderby cust.LastName
                  select new MyCustomer
                  {
                    LastName = cust.LastName,
                    FirstName = cust.FirstName
                  };

      return items;
    }
  }
}
